src/app/request/request.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RequestService {
+
+  constructor(private http: HttpClient) { }
+
+  public sendRequest(method: string, path: string, body?: any, headers?: HttpHeaders): Observable<any> {
```

Comment:
It's ok to have generic methods to adhere to DRY, but you should expose meaningful methods to other components. The meaningful methods can use the generic methods under the hood.